File: atv_professores/src/ProfessorHorista.java
public class ProfessorHorista extends Professor {
    int horasTrabalhadas;

    static final double VALOR_HORA = 25.00;

    @Override
    public void lecionar() {
        System.out.println("Professor horista está passando aula!");
    }

    @Override
    public double getSalario() {
        return this.horasTrabalhadas * ProfessorHorista.VALOR_HORA;
    }

    public int getHorasTrabalhadas() {
        return horasTrabalhadas;
    }

    public void setHorasTrabalhadas(int horasTrabalhadas) {
        this.horasTrabalhadas = horasTrabalhadas;
    }
}

File: atv_professores_kotlin/src/main/kotlin/ProfessorHorista.kt
class ProfessorHorista : Professor() {
    var horasTrabalhadas = 0

    override fun lecionar() {
        println("Professor horista está passando aula!")
    }

    override var salario: Double
        get() = horasTrabalhadas * VALOR_HORA
        set(salario) {
            super.salario = salario
        }

    companion object {
        const val VALOR_HORA = 25.00
    }
}
